cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -907,45 +871,67 @@ func (p *BicepProvider) modulePath() string {
 }
 
 // Ensures the provisioning parameters are valid and prompts the user for input as needed
-func (p *BicepProvider) ensureParameters(ctx context.Context, deployment *Deployment) (bool, error) {
+func (p *BicepProvider) ensureParameters(
+	ctx context.Context,
+	asyncContext *async.InteractiveTaskContextWithProgress[*DeploymentPlan, *DeploymentPlanningProgress],
+	deployment *Deployment,
+) error {
 	if len(deployment.Parameters) == 0 {
-		return false, nil
+		return nil
 	}
 
-	updatedParameters := false
 	for key, param := range deployment.Parameters {
 		// If this parameter has a default, then there is no need for us to configure it
 		if param.HasDefaultValue() {
 			continue
 		}
 		if !param.HasValue() {
-			userValue, err := p.console.Prompt(ctx, input.ConsoleOptions{
-				Message: fmt.Sprintf("Please enter a value for the '%s' deployment parameter:", key),
-			})
+			configKey := fmt.Sprintf("infra.parameters.%s", key)
 
-			if err != nil {
-				return false, fmt.Errorf("prompting for deployment parameter: %w", err)
+			if v, has := p.env.Config.Get(configKey); has {
+				param.Value = v
+				deployment.Parameters[key] = param
+				continue
 			}
 
-			param.Value = userValue
+			if err := asyncContext.Interact(func() error {
+				userValue, err := p.console.Prompt(ctx, input.ConsoleOptions{
+					Message: fmt.Sprintf("Please enter a value for the '%s' deployment parameter:", key),
+				})
 
-			saveParameter, err := p.console.Confirm(ctx, input.ConsoleOptions{
-				Message: "Save the value in the environment for future use",
-			})
+				if err != nil {
+					return fmt.Errorf("prompting for deployment parameter: %w", err)
+				}
 
-			if err != nil {
-				return false, fmt.Errorf("prompting to save deployment parameter: %w", err)
-			}
+				saveParameter, err := p.console.Confirm(ctx, input.ConsoleOptions{
+					Message: "Save the value in the environment for future use",
+				})
 
-			if saveParameter {
-				p.env.Values[key] = userValue
-			}
+				if err != nil {
+					return fmt.Errorf("prompting to save deployment parameter: %w", err)
+				}
 
-			updatedParameters = true
+				if saveParameter {
+					if err := p.env.Config.Set(configKey, userValue); err != nil {
+						return fmt.Errorf("saving value: %w", err)
+					}
+
+					if err := p.env.Save(); err != nil {
+						return fmt.Errorf("saving value: %w", err)
+					}
+				}
+
+				param.Value = userValue
+				deployment.Parameters[key] = param
+
+				return nil
+			}); err != nil {
+				return err
```

Comment:
nit. Consider to split the `if` and `interact` in two lines, in the for of:
err = interact(....)
if err != nil {}

The code inside the interact callback is long enough to disconnect the line 888 from 929 within the screen.
Or consider moving the callback for interact to be out and then just use it as param for calling interact, like
if err:=interact(callbac); err != nil { ... }